src/Workleap.Extensions.Mongo.Abstractions/MongoReflectionCache.cs

```diff
@@ -26,10 +32,26 @@ public static string GetCollectionName(Type documentType)
     }
 
     public static string GetCollectionName<TDocument>()
-        where TDocument : IMongoDocument
+        where TDocument : class
     {
         return GetCollectionName(typeof(TDocument));
     }
+    
+    internal static void SetCollectionName(Type documentType, string collectionName)
+    {
+        if (!CollectionNames.TryAdd(documentType, collectionName))
+        {
+            throw new ArgumentException($"Collection name for {documentType} already set.");            
+        }
+    }
+    
+    internal static void AddIndexProviderType(Type documentType, Type? indexProviderType)
```

Comment:
In the case of Index it is not really a caching since the indexes from the Attributes approach won't ever end here. Also it feel like we are adding additional responsability and lowering cohesion to this class: new new field that is not required nor used inside this class. I think we could create a new class specific for RegisteredIndexes,

Reply:
Refactored into a separate class